### PR TITLE
Write tests for cassandra/tasks.go

### PR DESCRIPTION
**Pull request by taylanisikdemir**
**What changed?**
Added unit tests for cassandra/tasks.go

**Why?**
Improve coverage

### CODE REVIEW

File: common/persistence/nosql/nosqlplugin/cassandra/tasks_test.go [hunk 1]
```diff
@@ -0,0 +1,1086 @@
...
+				LastUpdatedTime: ts,
+			},
+			queryMockFn: func(query *gocql.MockQuery) {
+				query.EXPECT().WithContext(gomock.Any()).Return(query).Times(1)
+				query.EXPECT().MapScanCAS(gomock.Any()).DoAndReturn(func(prev map[string]interface{}) (bool, error) {
+					return true, nil
+				}).Times(1)
+			},
+			wantQueries: []string{
+				`UPDATE tasks SET range_id = 25, task_list = {domain_id: domain1, name: tasklist1, type: 1, ack_level: 1000, kind: 2, last_updated: 2024-04-01T22:08:41Z } WHERE domain_id = domain1 and task_list_name = tasklist1 and task_list_type = 1 and type = 1 and task_id = -12345 IF range_id = 25`,
```
Comment (davidporter-id-au) [inline common/persistence/nosql/nosqlplugin/cassandra/tasks_test.go:281]: having these document is really nice from a readability/documentation perspective

File: common/persistence/nosql/nosqlplugin/cassandra/tasks_test.go [hunk 2]
```diff
@@ -0,0 +1,1086 @@
...
+			},
+			tasklistCond: &nosqlplugin.TaskListRow{
+				DomainID:     "domain1",
+				TaskListName: "tasklist1",
+				TaskListType: 1,
+				RangeID:      25,
+			},
+			mapExecuteBatchCASApplied: true,
+			wantQueries: []string{
+				`INSERT INTO tasks (domain_id, task_list_name, task_list_type, type, task_id, task) VALUES(domain1, tasklist1, 1, 0, 3, {domain_id: domain1, workflow_id: wid1, run_id: rid1, schedule_id: 42,created_time: 0001-01-01T00:00:00Z, partition_config: map[] })`,
```
Comment (davidporter-id-au) [inline common/persistence/nosql/nosqlplugin/cassandra/tasks_test.go:672]: the `created_time`, is that expected to be zero?

**Review by davidporter-id-au — APPROVED** [review]

**Comment by davidporter-id-au** [comment]
this snapshotting of real data is really good for readability and understanding what the code does in action

File: common/persistence/nosql/nosqlplugin/cassandra/tasks_test.go (hunk 2 above)
Comment (taylanisikdemir) [reply common/persistence/nosql/nosqlplugin/cassandra/tasks_test.go:672]: I missed to pass `TaskRow.CreatedTime`. Good catch let me add that

**Pull request merged**